make run for native_posix has dependency issues

call 'make -j10 run' for native_posix and the build will fail, we do not have dependency of the run taget on the .exe binary.

```
Scanning dependencies of target run
make[3]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make -f zephyr/CMakeFiles/run.dir/build.make zephyr/CMakeFiles/run.dir/build
make[3]: Entering directory '/home/nashif/Work/github/zephyr/build'
/home/nashif/Work/github/zephyr/build/zephyr/zephyr.exe
make[3]: /home/nashif/Work/github/zephyr/build/zephyr/zephyr.exe: Command not found
make[3]: *** [zephyr/CMakeFiles/run.dir/build.make:60: zephyr/CMakeFiles/run] Error 127
make[3]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make[2]: *** [CMakeFiles/Makefile2:271: zephyr/CMakeFiles/run.dir/all] Error 2
make[2]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make[1]: *** [CMakeFiles/Makefile2:278: zephyr/CMakeFiles/run.dir/rule] Error 2
make[1]: Leaving directory '/home/nashif/Work/github/zephyr/build'
make: *** [Makefile:173: run] Error 2
make: Leaving directory '/home/nashif/Work/github/zephyr/build'

```

--- a/cmake/emu/native.cmake
+++ b/cmake/emu/native.cmake
@@ -1,6 +1,7 @@
 add_custom_target(run
   COMMAND
   ${APPLICATION_BINARY_DIR}/zephyr/${KERNEL_EXE_NAME}
+  DEPENDS ${logical_target_for_zephyr_elf}
   WORKING_DIRECTORY ${APPLICATION_BINARY_DIR}
   USES_TERMINAL
   )

